src/azure-cli-testsdk/azure/cli/testsdk/utilities.py

```diff
@@ -34,3 +36,27 @@ def get_sha1_hash(file_path):
             sha1.update(data)
 
     return sha1.hexdigest()
+
+
+def find_recording_dir(test_file):
+    """ Find the directory containing the recording of given test file based on current profile. """
+    from azure.cli.core._profile import get_active_cloud, init_known_clouds
+    from azure.cli.core.cloud import CloudNotRegisteredException
+    try:
+        api_profile = get_active_cloud().profile
+    except CloudNotRegisteredException:
+        init_known_clouds()
+        api_profile = get_active_cloud().profile
```

Comment:
Yes. The az run can take care of CI. But it doesn't rule out the local environment when I some case I intentionally remove the profile folder.